updates values.yaml config.kubernetes to config.kubernetes_executor

N-ickJones opened the pull request:
closes: #29817 

When deploying Helm chart 1.8.0 the scheduler is expecting the airflow.cfg file to contain a kubernetes_executor section but, instead contains a kubernetes section. This change was introduced during the Airflow release v2.5.0 # `Rename kubernetes config section to kubernetes_executor` [#26873](https://github.com/apache/airflow/pull/26873)

jedcunningham reviewed (changes requested):
Thanks for the PR @N-ickJones!

Unfortunately, it's not quite as simple as your proposed change. The latest version of the chart still supports older Airflow versions. Check out what was done for the [logging config move](https://github.com/apache/airflow/blob/4b36137a31241d0f502604213546b6bf677fea69/chart/values.yaml#L1784-L1789).

jedcunningham commented on chart/values.yaml:
```diff
@@ -1830,6 +1830,14 @@ config:
+  kubernetes_executor:
+    namespace: '{{ .Release.Namespace }}'
+    airflow_configmap: '{{ include "airflow_config" . }}'
+    airflow_local_settings_configmap: '{{ include "airflow_config" . }}'
```
```suggestion
```

These 2 are for 1.10 and earlier versions, yeah? We definitely don't need to duplicate those!

N-ickJones replied:
I removes those 2 properties and added comments.

N-ickJones replied:
Thanks for the help. Hopefully, I updated the PR appropriately.

jedcunningham reviewed (changes requested):
Overall, looks good. A couple comments and we should be good to merge this!

jedcunningham commented on chart/values.yaml:
```diff
@@ -1822,6 +1822,8 @@ config:
     ccache: '{{ .Values.kerberos.ccacheMountPath }}/{{ .Values.kerberos.ccacheFileName }}'
   celery_kubernetes_executor:
     kubernetes_queue: 'kubernetes'
+  # The `kubernetes` section is deprecated in Airflow >= 2.5.0 due to a airflow.cfg scheme change. 
```
```suggestion
  # The `kubernetes` section is deprecated in Airflow >= 2.5.0 due to an airflow.cfg scheme change. 
```

jedcunningham commented on chart/values.yaml:
```diff
@@ -1822,6 +1822,8 @@ config:
+  # The `kubernetes` section can be removed once the helm chart no longer supports Airflow < 2.5.0.
   kubernetes:
     namespace: '{{ .Release.Namespace }}'
     airflow_configmap: '{{ include "airflow_config" . }}'
```
```suggestion
    # The following `airflow_` entries are for Airflow 1, and can be removed when it is no longer supported.
    airflow_configmap: '{{ include "airflow_config" . }}'
```

Let's add a note about these as well.

N-ickJones replied:
note added

N-ickJones replied:
The recent force pushes are to squash commits, clean the commit message, and update the fork to the latest of airflow main.

N-ickJones committed:
update values.yaml `config.kubernetes_executor`

For backwards compatibility of Airflow <= 2.5.0
keeps the existing values.yaml `config.kubernetes`
and add a new `config.kubernetes_executor` section.

values.yaml `config.kubernetes` deprecation in Airflow >= 2.5.0.
values.yaml `config.kubernetes` deprecation of `airflow_` entries.

jedcunningham commented on chart/values.yaml:
```diff
@@ -1888,14 +1888,24 @@ config:
     ccache: '{{ .Values.kerberos.ccacheMountPath }}/{{ .Values.kerberos.ccacheFileName }}'
   celery_kubernetes_executor:
     kubernetes_queue: 'kubernetes'
+  # The `kubernetes` section is deprecated in Airflow >= 2.5.0 due to an airflow.cfg schema change. 
```
```suggestion
  # The `kubernetes` section is deprecated in Airflow >= 2.5.0 due to an airflow.cfg schema change.
```

jedcunningham committed:
Update chart/values.yaml
